crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3089,13 +3120,20 @@ pub enum KnownFunction {
 
     /// [`typing(_extensions).no_type_check`](https://typing.readthedocs.io/en/latest/spec/directives.html#no-type-check)
     NoTypeCheck,
+
+    /// Type API functions (`knot_extensions`)
+    TypeApiFunction(TypeApiFunction),
```

Comment:
Similarly, I wonder why we need `TypeApiFunction` nested enum here, rather than including the type-api functions as top-level known functions in `KnownFunction` enum?

Reply:
It allows me to exhaustively match on all type API functions later without having to use `unreachable!`. But if we're all okay with a tighter integration of this code here, I can see if it makes sense to flatten this.

Reply:
> It allows me to exhaustively match on all type API functions later

Right. I think the idea of flattening is predicated on the idea that we _shouldn't_ have any place where we match on "all type API functions" (as opposed to "all known functions with special-cased call handling").

Reply:
Ok, the type API functions are now handled in the same places like `RevealType` etc.